.evergreen/print-compass-env.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+function maybePrependPath(path, part) {
+  const substring = `${part}:`;
+  if (path.includes(substring)) {
+    return path;
+  }
+  return `${substring}${path}`;
```

Comment:
This changes the logic a bit – are you sure you want this “maybe” bit here? The order of entries in `PATH` matters a lot, and having duplicates is fine, so it seems like it would be more reliable to always prepend entries?

Reply:
I know it matters and that duplicates are fine, but I want this script to be idempotent and only modify the path the first time you run it.

Reply:
Then maybe you could join all the PATH prepends into a single string, and check whether PATH already starts with that? That way you still fulfill that requirement but still have a solution that gives you a consistent order in the result

Reply:
Yeah, that’s the point – you check whether `$PATH` starts with `C:B:A:` and prepend `C:B:A:` it if it does not. That way the script is fully idempotent, but you also don’t end up in the situation that `X:Y:A:Z` gets turned into `X:Y:A:Z` but `X:Y:Z` gets turned into `A:X:Y:Z`, where the relative priority of `X` and `A` depends on whether `A` was already in `$PATH` (which might not seem like a big deal, but I would be afraid that that leads to hard-to-debug issues in CI otherwise, where for example `A` might already be in `$PATH` on some machines but not others).

Reply:
I'll change it but I think this is all unnecessary because I'm in control of this script that I wrote.